agent/consul/server.go

```diff
@@ -728,10 +728,11 @@ func NewServer(config *Config, flat Deps, externalGRPCServer *grpc.Server) (*Ser
 	s.externalConnectCAServer.Register(s.externalGRPCServer)
 
 	dataplane.NewServer(dataplane.Config{
-		GetStore:    func() dataplane.StateStore { return s.FSM().State() },
-		Logger:      logger.Named("grpc-api.dataplane"),
-		ACLResolver: s.ACLResolver,
-		Datacenter:  s.config.Datacenter,
+		GetStore:                          func() dataplane.StateStore { return s.FSM().State() },
+		Logger:                            logger.Named("grpc-api.dataplane"),
+		ACLResolver:                       s.ACLResolver,
+		Datacenter:                        s.config.Datacenter,
+		MergeNodeServiceWithCentralConfig: mergeNodeServiceWithCentralConfig,
```

Comment:
I'm wondering how much work it'd be to move `mergeNodeServiceWithCentralConfig` into `agent/configentry`.

Aside from the `state.Store` parameter, I can't immediately see anything that'd create an import cycle (and we can swap that parameter for an interface).

We'll also need this method to merge config for xDS, so I think moving it is worthwhile.

Reply:
You are right! Moving this was easier than I thought: https://github.com/hashicorp/consul/pull/14869/commits/9935ac249523d40e5499a23a5b89cb457ae6b0b8

It's now at `configentry.MergeNodeServiceWithCentralConfig` and it accepts an interface for the state store like you suggested. A lot cleaner 🙏